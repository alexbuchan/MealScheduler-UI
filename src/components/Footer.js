import React from 'react';

class Footer extends React.Component {
  render() {
    return (
      <footer className="footer navbar-dark bg-dark">
        <div className="footer-copyright text-center py-3">Â© 2020 Copyright:
          <a href="https://github.com/alexbuchan"> alex.buchan@telefonica.net</a>
        </div>
      </footer>
    );
  }
}

export default Footer;
