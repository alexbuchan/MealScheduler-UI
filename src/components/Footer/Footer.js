import React from 'react';

class Footer extends React.Component {
  render() {
    return (
      <footer className="footer">
        <div className="footer-copyright">Â© 2020 Copyright:
          <a href="https://github.com/alexbuchan"> alex.buchan@telefonica.net</a>
        </div>
      </footer>
    );
  }
}

export default Footer;
