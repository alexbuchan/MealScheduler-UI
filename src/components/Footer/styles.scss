.footer {
  position: fixed; 
  bottom: 0;
  width: 100%;
  background-color: $headerAndFooterColor;
  color: rgba(255,255,255,.5);
  display: flex;
  justify-content: center;
  align-items: center;
  height: $footerHeight;
}

.footer-copyright {
  color: #d6d6d6;
  font-size: smaller;
}

@media only screen and (max-width: 649px) {
  .footer {
    position: fixed; 
    bottom: 0;
    width: 100%;
    background-color: $headerAndFooterColor;
    color: rgba(255,255,255,.5);
    display: flex;
    justify-content: center;
    align-items: center;
    height: 60px;
  }
  
  .footer-copyright {
    color: #d6d6d6;
    font-size: smaller;
  }
}