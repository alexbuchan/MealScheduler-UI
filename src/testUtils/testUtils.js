import React from 'react';
import { shallow } from 'enzyme';

export const componentSetup = (Component, props = {}, options = {}) => {
  const instance = shallow(<Component { ...props } />, options);
  return instance;
}

export const findByTestAttribute = (instance, attribute) => {
  const wrapper = instance.find(`[data-test="${attribute}"]`);
  return wrapper;
}