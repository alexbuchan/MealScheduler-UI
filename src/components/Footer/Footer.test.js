import Footer from './Footer';
import { componentSetup } from '../../testUtils/testUtils';

describe('Footer', () => {
  let instance;
  
  beforeEach(() => {
    instance = componentSetup(Footer);
  });

  it('Should render without errors', () => {
    expect(instance).toMatchSnapshot();
  });
});