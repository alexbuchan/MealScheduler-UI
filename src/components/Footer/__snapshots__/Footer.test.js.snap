// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should render without errors 1`] = `
<footer
  className="footer"
  data-test="footer"
>
  <div
    className="footer-copyright"
  >
    Â© 2020 Copyright:
    <a
      data-test="author-link"
      href="https://github.com/alexbuchan"
    >
       alex.buchan@telefonica.net
    </a>
  </div>
</footer>
`;
